uber/site_sections/registration.py

```diff
@@ -885,6 +885,113 @@ def discount(self, session, message='', **params):
 
         return {'message': message}
 
+    @csv_file
+    def generate_promo_codes(self, out, session, **params):
```

Comment:
This function is getting printed to the site map and is throwing an error if you try to load it.

In general, the code to generate the promo codes doesn't feel finished. You have:
1) a template called `generate_batch_codes.html` that looks like a copy of `view_promo_codes.html` and doesn't have a corresponding function.
2) a template/function pair that loads a form to generate 1 or more promo codes and then...
3) calls a different function to print out a CSV.

It just doesn't make a lot of sense. From a UX perspective, it's confusing as well - I click the button to generate codes and just get a CSV file. I didn't ask for a CSV file. I asked to generate codes -- and I get no confirmation of the system having done so, except for a CSV file I didn't want.

You definitely should get rid of the extraneous template. As for the UX, it will be a lot better if a user can click the 'generate' function and have the page simply refresh with a list of the promo codes they generated, and a button to get a CSV of those codes.